infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/pvc/SubPathPrefixes.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.namespace.pvc;
+
+import static org.eclipse.che.workspace.infrastructure.kubernetes.Constants.CHE_VOLUME_NAME_LABEL;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.provision.LogsVolumeMachineProvisioner.LOGS_VOLUME_NAME;
+
+import com.google.common.base.Strings;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaim;
+import io.fabric8.kubernetes.api.model.VolumeMount;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.eclipse.che.workspace.infrastructure.kubernetes.Names;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment;
+import org.eclipse.che.workspace.infrastructure.kubernetes.environment.KubernetesEnvironment.PodData;
+
+/**
+ * Is responsible for prefixing sub-paths of volume mounts and should be used by all PVCs
+ * strategies.
+ *
+ * @author Sergii Leshchenko
+ */
+public class SubPathPrefixes {
+
+  /**
+   * Prefixes volumes mounts of containers inside of the specified kubernetes environment.
+   *
+   * <p>Subpaths have the following format: '{workspaceId}/{Che Volume name|PVC name}'.<br>
+   * Where Che Volume is used if it is present in PVC labels, otherwise PVC name will be used.<br>
+   * Note that logs volume has the special format: '{workspaceId}/{volumeName}/{machineName}'. It is
+   * done in this way to avoid conflicts e.g. two identical agents inside different machines produce
+   * the same log file.
+   *
+   * @param k8sEnv environment to process
+   * @param workspaceId workspace id that should be used as prefix
+   */
+  public void prefixVolumeMountsSubpaths(KubernetesEnvironment k8sEnv, String workspaceId) {
```

Comment:
Same as `PodVolumes`, this class could be static.

Reply:
This class cannot be static because of unit tests. feel free to reopen comment if you have any ideas on how to implement mock testing with static methods without `PowerMock`